File: Go/Find-The-Highest-Altitude.go
package main

func largestAltitude(gain []int) int {

	currentAltitude, maxAltitude := 0, 0

	for _, change := range gain {
		currentAltitude += change
		maxAltitude = max(maxAltitude, currentAltitude)
	}

	return maxAltitude
}

File: Rust/Find-The-Highest-Altitude.rs
impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut current_altitude = 0;
        let mut max_altitude = 0;

        for change in gain {
            current_altitude += change;
            max_altitude = std::cmp::max(max_altitude, current_altitude)
        }

        max_altitude
    }
}

// 28.06.2023